### PR TITLE
FEAT-327 Move minio to hbw

### PR DESCRIPTION
**Pull request by GideonEXE**

### CODE REVIEW

File: hbw/app/assets/javascripts/hbw/components/form/file_list.js.jsx.coffee [hunk 1]
```diff
@@ -2,25 +2,44 @@ modulejs.define 'HBWFormFileList', ['React', 'HBWDeleteIfMixin'], (React, Delete
...
+            {this.files(this.state.links)}
           </ul>
         </label>
       </div>`
 
     files: (list) ->
+      self = this
       list.map (variant) ->
-        `<li><a href={variant.url}>{variant.name}</a></li>`
+        `<li><a href={variant.url}>{variant.name}</a>&nbsp;<a href="#" className="fa fa-times" onClick={ e => self.deleteLink(e, variant.name)}></a></li>`
```
Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw/components/form/file_list.js.jsx.coffee]:
```
onClick={ e
```

File: hbw/app/controllers/hbw/tasks_controller.rb [hunk 2]
```diff
@@ -19,7 +26,27 @@ def edit
     end
 
     def submit
-      result = widget.submit(current_user.email, entity_class, task_id, form_data)
+      files = form_data.map{ |key, value| JSON.parse(value) rescue nil }
```
Comment (sugarfree1) [inline hbw/app/controllers/hbw/tasks_controller.rb]: We put space before blocks:
```form_data.map{```
to
```form_data.map {```

File: hbw/app/controllers/hbw/tasks_controller.rb [hunk 3]
```diff
@@ -19,7 +26,27 @@ def edit
...
+                       .compact
+                       .map{ |value| value['files'] if value.respond_to?(:key) }
+                       .compact.flatten
+
+      fields_for_save = form_data.map{ |key, value| JSON.parse(value)['files'] ? nil : key  rescue key }.compact
```
Comment (sugarfree1) [inline hbw/app/controllers/hbw/tasks_controller.rb]: This block contains complex logic. Better use do/end notation.

File: hbw/app/controllers/hbw/tasks_controller.rb [hunk 4]
```diff
@@ -19,7 +26,27 @@ def edit
...
+
+      data = form_data.select{ |key, value| fields_for_save.include?(key)}
```
Comment (sugarfree1) [inline hbw/app/controllers/hbw/tasks_controller.rb]: Space at the end of block.

File: hbw/lib/minio/adapter.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      result = []
+      unless files.nil?
+        files.each do |file|
+          s3 = Aws::S3::Resource.new(Aws::S3::Client.new)
+          bucket = s3.bucket(HBW::Widget.config[:minio][:bucket])
+          file_name = "#{task_id}_#{file['name']}"
+          obj = bucket.object(file_name)
+          obj.put(body: Base64.decode64(file['content']))
+
+
```
Comment (sugarfree1) [inline hbw/lib/minio/adapter.rb]: Two new lines.

**Comment by GideonEXE** [comment]
UPD

File: hbw/app/assets/javascripts/hbw/components/form/file_list.js.jsx.coffee [hunk 6]
```diff
@@ -2,25 +2,43 @@ modulejs.define 'HBWFormFileList', ['React', 'HBWDeleteIfMixin'], (React, Delete
...
           <span>{' ' + label}</span>
           <ul>
-            {this.files(links)}
+            {this.files(this.state.links)}
           </ul>
         </label>
       </div>`
 
     files: (list) ->
+      self = this
```
Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw/components/form/file_list.js.jsx.coffee]: You can save ```this.deleteLink``` to local var. That's more obvious.

File: hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee [hunk 7]
```diff
@@ -5,14 +5,19 @@ modulejs.define 'HBWFormFileUpload',
...
+        {value: '', valid: true, files: [], files_count: 0, incoming_files: []}
 
       render: ->
         opts = {
           name: @props.params.name
           onChange: @onChange
         }
 
+        if (@props.value)
+          @state.incoming_files = JSON.parse(@props.value).files
```
Comment (sugarfree1) [inline hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee]: ```props``` should not be modified. So, you set ```incoming_files``` every if you have value in props. Every time. Why?

File: hbw/app/controllers/hbw/tasks_controller.rb [hunk 8]
```diff
@@ -19,7 +26,31 @@ def edit
     end
 
     def submit
-      result = widget.submit(current_user.email, entity_class, task_id, form_data)
+      files = form_data.map { |key, value| JSON.parse(value) rescue nil }
```
Comment (sugarfree1) [inline hbw/app/controllers/hbw/tasks_controller.rb]: Seems like it's a job for service or saver.

File: hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee (hunk 7 above)
Comment (GideonEXE) [reply hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee]: props can be modified by activiti, imprint integrations for example.

Comment (sugarfree1) [reply hbw/app/assets/javascripts/hbw/components/form/file_upload.js.jsx.coffee]: Maybe I don't understand you, but https://reactjs.org/docs/components-and-props.html#props-are-read-only

**Comment by GideonEXE** [comment]
UPD

**Comment by GideonEXE** [comment]
UPD

**Pull request merged**